homeassistant/components/izone/update_coordinator.py

```diff
@@ -0,0 +1,186 @@
+"""Data update coordinator for iZone."""
+from collections.abc import Callable
+import logging
+from typing import Any
+
+import pizone
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo, Entity
+
+from .const import (
+    DISPATCH_CONTROLLER_DISCONNECTED,
+    DISPATCH_CONTROLLER_RECONNECTED,
+    IZONE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ControllerUpdateCoordinator:
+    """Coordinate updates for a controller. This works similarly to homeassistant.helpers.update_coordinator.DataUpdateCoordinator."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        ctrl: pizone.Controller,
+    ) -> None:
+        """Initialise the update coordinator."""
+        self._controller = ctrl
+        self._available_listeners: list[CALLBACK_TYPE] = []
+        self._available = True
+        self._device_info = DeviceInfo(
```

Comment:
Device info is an entity api, so we should not put it here in the coordinator. Put it in a common entity base class.

Reply:
Makes sense. Took it out.